src/aliceVision/feature/sift/ImageDescriber_DSPSIFT_vlfeat.cpp

```diff
@@ -0,0 +1,393 @@
+                filteredIndexes.insert(filteredIndexes.end(), rejectedIndexes.begin(),
+                                        rejectedIndexes.begin() + remainingElements);
+            }
+            indexSort.swap(filteredIndexes);
+            ALICEVISION_LOG_TRACE("SIFT Features: before: " << nbFeatures
+                                                            << ", after grid filtering: " << filteredIndexes.size());
+        }
+    }
+    else if(params._maxTotalKeypoints)
+    {
+        // Retrieve extracted features
+        if(indexSort.size() > params._maxTotalKeypoints)
+            indexSort.resize(params._maxTotalKeypoints);
+    }
+
+    // Compute the descriptors for the detected keypoints.
+    {
+        regionsCasted->Features().resize(indexSort.size());
+        regionsCasted->Descriptors().resize(indexSort.size());
+
+        std::unique_ptr<VlSiftFilt, void (*)(VlSiftFilt*)> sift(vl_sift_new(16, 16, 1, params._numScales, 0),
+                                                                &vl_sift_delete);
+
+        // All constant parameters
+        const size_t kPatchResolution = 15;
+        const size_t kPatchSide = 2 * kPatchResolution + 1;
+        const double kPatchRelativeExtent = 7.5;
+        const double kPatchRelativeSmoothing = 1;
+        const double kPatchStep = kPatchRelativeExtent / kPatchResolution;
+        const double kSigma = kPatchRelativeExtent / (3.0 * (4 + 1) / 2) / kPatchStep;
+        // DSP parameters
+        float dspMinScale = 1;
+        float dspScaleStep = 0;
+        int dspNumScales = 1;
+        if(params.domainSizePooling)
+        {
+            dspMinScale = params.dspMinScale;
+            dspScaleStep = (params.dspMaxScale - params.dspMinScale) / params.dspNumScales;
+            dspNumScales = params.dspNumScales;
+        }
+
+#pragma omp parallel
+        {
+            VlCovDetBuffer internalBuffer;
+            vl_covdetbuffer_init(&internalBuffer);
+            Eigen::Matrix<float, Eigen::Dynamic, 128, Eigen::RowMajor> descriptorsOverDspScales(dspNumScales, 128);
+            std::vector<float> patch(kPatchSide * kPatchSide);
+            std::vector<float> patchXY(2 * kPatchSide * kPatchSide);
+
+#pragma omp for
+            for(int oIndex = 0; oIndex < indexSort.size(); ++oIndex)
+            {
+                const int iIndex = indexSort[oIndex];
+                const auto& inFeat = features[iIndex];
+
+                for(int s = 0; s < dspNumScales; ++s)
+                {
+                    const double dspScale = dspMinScale + s * dspScaleStep;
+
+                    VlFrameOrientedEllipse frameAtScale = inFeat.frame;
+                    frameAtScale.a11 *= dspScale;
+                    frameAtScale.a12 *= dspScale;
+                    frameAtScale.a21 *= dspScale;
+                    frameAtScale.a22 *= dspScale;
+
+                    vl_covdet_extract_patch_for_frame(covdet.get(), patch.data(), kPatchResolution, &internalBuffer,
```

Comment:
To me, this is extracting a sub image at a given location at an interpolated scale.

I assume the scale interpolation is based on the existing, pre-computed scales. So this is bounded by the parameters of the sift detector. As such, shouldn't the scales be computed to make sure that for each keypoint, we are able to  get a range of scales large enough ?